org.mixedrealitytoolkit.input/InteractionModes/NearInteractionModeDetector.cs

```diff
@@ -25,49 +28,67 @@ public class NearInteractionModeDetector : ProximityDetector
+        /// effectively triggers OnProximityExited for all colliders that are no longer detected.
+        /// </summary>
+        private void UpdateProximityExited()
+        {
+            for (int i = 0; i < previouslyDetectedColliders.Count; i++)
+            {
+                Collider previouslyDetectedCollider = previouslyDetectedColliders[i];
+                if (!DetectedColliders.Contains(previouslyDetectedCollider) &&
+                    InteractionManager.TryGetInteractableForCollider(previouslyDetectedCollider, out IXRInteractable xrInteractable) &&
+                    xrInteractable is IXRProximityInteractable xrProximityInteractable)
                 {
-                    Collider previouslyDetectedCollider = previouslyDetectedColliders[i];
-                    if (!DetectedColliders.Contains(previouslyDetectedCollider) && previouslyDetectedCollider != null)
+                    foreach (XRBaseInteractor xrBaseInteractor in nearInteractors)
                     {
-                        IXRProximityInteractable nearInteractionMode = previouslyDetectedCollider.GetComponent<IXRProximityInteractable>();
-                        if (nearInteractionMode != null)
-                        {
-                            foreach (XRBaseInteractor xrBaseInteractor in nearInteractors)
-                            {
-                                previouslyDetectedCollider.GetComponent<IXRProximityInteractable>().OnProximityExited(new ProximityExitedEventArgs(this, previouslyDetectedCollider, xrBaseInteractor));
-                            }
-                        }
-                        previouslyDetectedColliders.Remove(previouslyDetectedCollider);
+                        xrProximityInteractable.OnProximityExited(new ProximityExitedEventArgs(previouslyDetectedCollider, xrBaseInteractor));
                     }
+                    previouslyDetectedColliders.Remove(previouslyDetectedCollider);
                 }
-                foreach (Collider collider in DetectedColliders)
+            }
+        }
+
+        /// <summary>
+        /// Calll OnProximityEntered for all colliders that are currently detected but were not detected previously.  This
+        /// effectively triggers OnProximityEntered for all colliders that are newly detected.
+        /// </summary>
+        private void UpdateProximityEntered()
+        {
+            foreach (Collider collider in DetectedColliders)
+            {
+                if (!previouslyDetectedColliders.Contains(collider) &&
+                    InteractionManager.TryGetInteractableForCollider(collider, out IXRInteractable xrInteractable) &&
+                    xrInteractable is IXRProximityInteractable xrProximityInteractable)
                 {
-                    if (!previouslyDetectedColliders.Contains(collider))
+                    foreach (XRBaseInteractor xrBaseInteractor in nearInteractors)
```

Comment:
I'm also still curious about your thoughts on https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/611#discussion_r1489878783

> Instead of sending multiple events to account for all `nearInteractors`, should we just send _one_ relaying info about either:
> 
> 1. The `ProximityDetector` itself. We could potentially expose the list of `nearInteractors` as an `IReadOnlyList` or something if we think the interactables need that info
> 2. Specifically the `nearInteractor`(s) doing the selecting. They likely aren't all selecting at once, but one of the cases for this logic is that _any_ `nearInteractor` is selecting (`IsNearInteractorSelecting()`)

Reply:
I agree, we can avoid this issue by instead storing Interactables instead of colliders.

Reply:
> I'm also still curious about your thoughts on [#611 (comment)](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/611#discussion_r1489878783)
> 
> > Instead of sending multiple events to account for all `nearInteractors`, should we just send _one_ relaying info about either:
> > 
> > 1. The `ProximityDetector` itself. We could potentially expose the list of `nearInteractors` as an `IReadOnlyList` or something if we think the interactables need that info
> > 2. Specifically the `nearInteractor`(s) doing the selecting. They likely aren't all selecting at once, but one of the cases for this logic is that _any_ `nearInteractor` is selecting (`IsNearInteractorSelecting()`)

At an earlier version of this PR my original approach was to filter the number of events risen by storing unique (Collider + Interactor) duples on the PressableButton side that were starting/stopping a ProximityHover events and only rise any such event when the corresponding collections would add or remove a unique duple.  Such approach is lost in translation at this point after implementing other feedback.

Reply:
> I agree, we can avoid this issue by instead storing Interactables instead of colliders.

Could you please elaborate more on why Interactables will prevent the issue?

Reply:
Switching from storing Interactor to storing Interactables in commit https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/commit/e2f21e5a5f09660f1e9db8569e5a2584692067d7